src/wallet/storage.rs

```diff
@@ -75,6 +85,29 @@ impl WalletStore {
         Ok(store)
     }
 
+    /// Internal function to update the UTXO cache from already-processed UTXOs.
+    pub(crate) fn update_utxo_cache_internal(&mut self, processed_utxos: Vec<(ListUnspentResultEntry, UTXOSpendInfo)>) {
```

Comment:
shorten the name `update_utxo_cache`. 

Also this should be a wallet API. You are getting the utxo list from `get_all_utxo` and passing the utxos to this function. That whole thing can be combined into one single function `wallet::update_utxo_cache(&mut self)`. Which will get all the utxos from rpc and update it. All in a single function. 

Then use that function at the end of the sync function.